### PR TITLE
ref(consumer): Remove batching strategy and associated consumer code

### PR DESCRIPTION
**Pull request by tkaemming**
This includes a collection of interrelated changes related to the removal of `ConsumerWorker`:

- `--strategy` has been removed from the command line interface, all consumers now use the streaming strategy (already the default since #1268)
- The `perf` tool has been removed, as it only tested the batching strategy
- `ConsumerWorker` and its associated tests have been removed
- The `eventstream` API endpoint now uses the streaming strategy
- The snapshot-aware consumer uses the streaming strategy by wrapping the storage processor in a `SnapshotProcessor` (this allows xid filtering to be parallelized) and removes the `SnapshotAwareWorker` shim

### CODE REVIEW

File: snuba/util.py [hunk 1]
```diff
@@ -220,20 +217,6 @@ def force_bytes(s: Union[bytes, str]) -> bytes:
...
-    previous = {}
-    for k, v in overrides.items():
-        previous[k] = getattr(settings, k, None)
-        setattr(settings, k, v)
-
-    try:
-        yield
-    finally:
-        for k, v in previous.items():
-            setattr(settings, k, v)
```
Comment (tkaemming) [inline snuba/util.py:234]: This was only used in perf.

File: snuba/web/views.py [hunk 2]
```diff
@@ -445,18 +445,32 @@ def eventstream(*, dataset: Dataset):
...
+                metrics,
+                max_batch_size=1,
+                max_batch_time=1.0,
+                processes=None,
+                input_block_size=None,
+                output_block_size=None,
+            ).create(lambda offsets: None)
+            strategy.submit(message)
+            strategy.close()
+            strategy.join()
```
Comment (tkaemming) [inline snuba/web/views.py:465]:
```
ted@veneno % py.test tests/snuba
21:30:43 [WARNING] sentry.utils.geo: settings.GEOIP_PATH_MMDB not configured.
/Users/ted/workspaces/sentry/sentry/src/sentry/utils/services.py:94: UnsupportedBackend: The 'sentry.digests.backends.dummy.DummyBackend' backend for <class 'sentry.digests.backends.base.Backend'> is not recommended for production use.
  "for production use.".format(self._backend, self._base)
21:30:49 [INFO] sentry.plugins.github: apps-not-configured
================================================================================================= test session starts ==================================================================================================
platform darwin -- Python 2.7.16, pytest-4.6.5, py-1.9.0, pluggy-0.13.1
rootdir: /Users/ted/workspaces/sentry/sentry, inifile: setup.cfg
plugins: django-3.5.1, metadata-1.10.0, celery-4.4.7, cov-2.5.1, sentry-0.1.5, html-1.22.0, rerunfailures-8.0
collected 626 items                                                                                                                                                                                                    

tests/snuba/test_snuba.py .....                                                                                                                                                                                  [  0%]
tests/snuba/test_util.py ...                                                                                                                                                                                     [  1%]
tests/snuba/api/endpoints/test_discover_key_transactions.py .............................                                                                                                                        [  5%]
tests/snuba/api/endpoints/test_discover_query.py ......x................                                                                                                                                         [  9%]
tests/snuba/api/endpoints/test_discover_saved_queries.py ....................                                                                                                                                    [ 12%]
tests/snuba/api/endpoints/test_discover_saved_query_detail.py ...........                                                                                                                                        [ 14%]
tests/snuba/api/endpoints/test_group_details.py ....                                                                                                                                                             [ 15%]
tests/snuba/api/endpoints/test_group_events.py ............                                                                                                                                                      [ 17%]
tests/snuba/api/endpoints/test_group_events_latest.py ...                                                                                                                                                        [ 17%]
tests/snuba/api/endpoints/test_group_events_oldest.py ...                                                                                                                                                        [ 18%]
tests/snuba/api/endpoints/test_organization_event_details.py ........                                                                                                                                            [ 19%]
tests/snuba/api/endpoints/test_organization_eventid.py ...                                                                                                                                                       [ 19%]
tests/snuba/api/endpoints/test_organization_events.py ................                                                                                                                                           [ 22%]
tests/snuba/api/endpoints/test_organization_events_facets.py ..................                                                                                                                                  [ 25%]
tests/snuba/api/endpoints/test_organization_events_measurements_histogram.py .......................x..x...                                                                                                      [ 30%]
tests/snuba/api/endpoints/test_organization_events_meta.py ......................                                                                                                                                [ 33%]
tests/snuba/api/endpoints/test_organization_events_stats.py ..........................................                                                                                                           [ 40%]
tests/snuba/api/endpoints/test_organization_events_trends.py ..................                                                                                                                                  [ 43%]
tests/snuba/api/endpoints/test_organization_events_v2.py ...................................................................................                                                                     [ 56%]
tests/snuba/api/endpoints/test_organization_group_index.py ..................................................................                                                                                    [ 66%]
tests/snuba/api/endpoints/test_organization_issues_resolved_in_release.py ......                                                                                                                                 [ 67%]
tests/snuba/api/endpoints/test_organization_tagkey_values.py .................                                                                                                                                   [ 70%]
tests/snuba/api/endpoints/test_organization_tags.py ......                                                                                                                                                       [ 71%]
tests/snuba/api/endpoints/test_project_event_details.py ........                                                                                                                                                 [ 72%]
tests/snuba/api/endpoints/test_project_events.py ...                                                                                                                                                             [ 73%]
tests/snuba/api/endpoints/test_project_group_index.py ..................................................                                                                                                         [ 81%]
tests/snuba/api/endpoints/test_project_tags.py .                                                                                                                                                                 [ 81%]
tests/snuba/api/serializers/test_group.py ..................                                                                                                                                                     [ 84%]
tests/snuba/eventstream/test_eventstream.py ..                                                                                                                                                                   [ 84%]
tests/snuba/incidents/test_tasks.py .                                                                                                                                                                            [ 84%]
tests/snuba/models/test_group.py .                                                                                                                                                                               [ 84%]
tests/snuba/search/test_backend.py ................................................x                                                                                                                             [ 92%]
tests/snuba/sessions/test_sessions.py .......                                                                                                                                                                    [ 93%]
tests/snuba/snuba/test_query_subscription_consumer.py ...                                                                                                                                                        [ 94%]
tests/snuba/tagstore/test_tagstore_backend.py .....................                                                                                                                                              [ 97%]
tests/snuba/tasks/test_unmerge.py ....                                                                                                                                                                           [ 98%]
tests/snuba/tsdb/test_tsdb_backend.py ..........                                                                                                                                                                 [100%]

======================================================================================= 622 passed, 4 xfailed in 1111.93 seconds =======================================================================================
py.test tests/snuba  225.13s user 18.04s system 21% cpu 18:42.33 total
```

File: snuba/web/views.py [hunk 3]
```diff
@@ -445,18 +445,32 @@ def eventstream(*, dataset: Dataset):
...
+                stream_loader.get_pre_filter(),
+                stream_loader.get_processor(),
+                table_writer.get_batch_writer(metrics),
+                metrics,
+                max_batch_size=1,
+                max_batch_time=1.0,
+                processes=None,
+                input_block_size=None,
+                output_block_size=None,
+            ).create(lambda offsets: None)
```
Comment (tkaemming) [inline snuba/web/views.py:462]: This is dummy callback is kind of a hack.

File: tests/test_consumer.py [hunk 4]
```diff
@@ -1,126 +1,22 @@
...
-        event["data"]["received"] = int(calendar.timegm(old_timestamp.timetuple()))
-
-        message: Message[KafkaPayload] = Message(
-            Partition(Topic("events"), 1),
-            42,
-            KafkaPayload(None, json.dumps((2, "insert", event)).encode("utf-8"), []),
-            datetime.now(),
-        )
-
-        assert test_worker.process_message(message) is None
```
Comment (tkaemming) [inline tests/test_consumer.py:97]: This was like three layers of abstraction in the wrong place. If it's important to keep around, it should be moved to the processor tests.

**Review by fpacifici — COMMENTED** [review]
Did you test the consumers still start? Afaik there is no test that covers the initialization process in consumer_builder.

File: snuba/cli/perf.py [hunk 5]
```diff
@@ -1,75 +0,0 @@
-"""\
```
[the remark this answers is not in the document]
Comment (tkaemming) [reply snuba/cli/perf.py:1]: No, not to my knowledge. If I recall correctly, it was also broken for a while earlier this year and nobody noticed (I might be thinking of something else though.)

**Comment by tkaemming** [comment]
> Did you test the consumers still start? Afaik there is no test that covers the initialization process in consumer_builder.

```
(.venv) ted@veneno % snuba consumer --log-level debug
2020-10-12 15:01:26,078 Starting
2020-10-12 15:01:28,009 New partitions assigned: {Partition(topic=Topic(name='events'), index=0): 5075475}
2020-10-12 15:01:28,010 Initialized processing strategy: <snuba.utils.streams.processing.strategies.streaming.transform.TransformStep object at 0x10cf97430>
^C2020-10-12 15:01:32,094 Shutdown signalled
2020-10-12 15:01:32,095 Stopping consumer
2020-10-12 15:01:32,095 Partitions revoked: [Partition(topic=Topic(name='events'), index=0)]
2020-10-12 15:01:32,095 Closing <snuba.utils.streams.processing.strategies.streaming.transform.TransformStep object at 0x10cf97430>...
2020-10-12 15:01:32,095 Waiting for <snuba.utils.streams.processing.strategies.streaming.transform.TransformStep object at 0x10cf97430> to exit...
2020-10-12 15:01:32,095 <snuba.utils.streams.processing.strategies.streaming.transform.TransformStep object at 0x10cf97430> exited successfully, releasing assignment.
2020-10-12 15:01:32,101 Stopped
```

**Review by fpacifici — APPROVED** [review]

**Pull request merged**